query-service-impl/src/test/java/org/hypertrace/core/query/service/postgres/QueryRequestToPostgresSQLConverterTest.java

```diff
@@ -0,0 +1,1212 @@
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "AND duration_millis = 1.2",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithIntFilter() {
+    QueryRequest queryRequest =
+        buildSimpleQueryWithFilter(createEqualsFilter("Span.metrics.duration_millis", 1));
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        queryRequest,
+        "Select encode(span_id, 'hex') FROM public.\"span-event-view\" WHERE "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "AND duration_millis = 1",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithTimestampFilter() {
+    QueryRequest queryRequest =
+        buildSimpleQueryWithFilter(createTimestampFilter("Span.is_entry", Operator.EQ, 123456));
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        queryRequest,
+        "Select encode(span_id, 'hex') FROM public.\"span-event-view\" WHERE "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "AND is_entry = 123456",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithOrderBy() {
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        buildOrderByQuery(),
+        "Select encode(span_id, 'hex'), start_time_millis, end_time_millis FROM public.\"span-event-view\" WHERE "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "order by start_time_millis desc , end_time_millis limit 100",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithOrderByWithPagination() {
+    QueryRequest orderByQueryRequest = buildOrderByQuery();
+    Builder builder = QueryRequest.newBuilder(orderByQueryRequest);
+    builder.setOffset(1000);
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        builder.build(),
+        "Select encode(span_id, 'hex'), start_time_millis, end_time_millis FROM public.\"span-event-view\" WHERE "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "order by start_time_millis desc , end_time_millis limit 1000, 100",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithGroupByWithMultipleAggregates() {
+    QueryRequest orderByQueryRequest = buildMultipleGroupByMultipleAggQuery();
```

Comment:
nit : `orderByQueryRequest` -> `groupByQueryRequest`

No order by used in this test.

Reply:
Done